### DISCUSSION TITLE
The specified resource does not exist - There is no resource of type 'AppsV1' with id 'APP_ID'

### DISCUSSION THREAD

**Discussion by demolaf** [codemagic-ci-cd/codemagic-docs#2078] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

**I keep getting the error below and I also see this in the Codemagic UI when deploying to Testflight**

**Error**:
GET https://api.appstoreconnect.apple.com/v1/apps/APP_ID/appStoreVersions?limit=100 returned 404: The specified resource does not exist - There is no resource of type 'AppsV1' with id 'APP_ID'

**I tried running this command to see if I could retrieve info from the appstore**

**Command**:
app-store-connect get-latest-build-number 'APP_ID' --issuer-id 'ISSUER_ID' --key-id 'KEY_ID'

**This script is meant to fetch the latest build number from app store connect and increment by 1**
Not sure if it works, maybe I'm missing something from the docs?

**Script**
name: Flutter build ipa and automatic versioning
script: |
flutter build ipa --release \
    --target lib/main.dart \
    --build-name=0.0.1 \
    --build-number=$(($(app-store-connect get-latest-app-store-build-number "$APP_ID") + 1))

Would appreciate feedback, thank you.

### Expected results

It should push the build to testflight but fails with an error.

### Actual results

"message": "The provided entity includes an attribute with a value that has already been used"
"iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE"
"NSLocalizedFailureReason": "The bundle version must be higher than the previously uploaded version"

**Comment by dtrdic** [comment]
Hey @demolaf, you can use this command for building your flutter ipa file and 

```
  LATEST_BUILD_NUMBER=$(($(app-store-connect get-latest-app-store-build-number "$APP_ID") + 1))
  flutter build ipa --release \
    --build-name=1.0.$LATEST_BUILD_NUMBER \
    --build-number=$LATEST_BUILD_NUMBER
```

or instead of fetching the lastest uploaded version from app store you can use the built-in env variable `BUILD_NUMBER`to setup versioning number
```
    agvtool new-version -all $(($BUILD_NUMBER + 1))
    flutter build ipa --release --flavor free \
      --build-name=1.0.$BUILD_NUMBER \
      --build-number=$BUILD_NUMBER \
      --export-options-plist=/Users/builder/export_options.plist
```

Here's the step-by-step [flutter guide](https://docs.codemagic.io/yaml-quick-start/building-a-flutter-app/) you can check.
Check out our [example workflow ](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/flutter/flutter-android-and-ios-yaml-demo-project/codemagic.yaml#L51)as well.

**Reply by demolaf** [reply]
Hey @dtrdic Thanks for the feedback, appreciate it.

**I'm still getting the initial error I was facing while trying to fetch the build-number of the latest build in testflight**

"NSLocalizedDescription": "The provided entity includes an attribute with a value that has already been used",
"previousBundleVersion": "14",
"iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE",
 "NSLocalizedFailureReason": "The bundle version must be higher than the previously 

**Here's a factor, I'm yet to consider:**
The latest build in testflight is an expired build with version number 14.
Could it be failing because it's an expired build?

Here's the script:
script: |
          agvtool new-version -all $(($BUILD_NUMBER + 1))
          flutter build ipa --release \
            --flavor production \
            --target lib/main_production.dart \
            --build-name=0.0.1 \
            --build-number=$BUILD_NUMBER \
            --export-options-plist=/Users/builder/export_options.plist

Is this line "agvtool new-version -all $(($BUILD_NUMBER + 1))" meant to increment the build number?

**Script from your solution:**
agvtool new-version -all $(($BUILD_NUMBER + 1))
    flutter build ipa --release --flavor free \
      --build-name=1.0.$BUILD_NUMBER \
      --build-number=$BUILD_NUMBER \
      --export-options-plist=/Users/builder/export_options.plist

If I use the above ( with my arguments of course ) I get an error with the "**--build-name**" argument.
**It says:**
CFBundleShortVersionString '0.0.1.4' in the Info.plist file at 'Payload\/Runner.app\/Frameworks\/sign_in_with_apple.framework' must be a period-separated list of at most three non-negative integers.

Which I guess is correct because build name can only consist of 3 non-negative integers as apple suggests.

Thanks again for the feedback, can I get a bit more assistance with this?

**Reply by dtrdic** [reply]
hey @demolaf how does look your property definition inside info.plist for the CFBundleShortVersionString key? does it look like this?
```
	<key>CFBundleShortVersionString</key>
	<string>$(FLUTTER_BUILD_NAME)</string>
```
and please share your build id so I can check it closer

**Reply by demolaf** [reply]
Thanks for the reply.

<img width="363" alt="Screenshot 2023-10-03 at 12 24 34" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/48495111/490c3256-8f81-4fcc-8516-520ae0d5d06a">

Yes it does, in the above screenshot attached.

Here's the build id from code magic: 65158595cf3ef888399c1b7a

**Reply by dtrdic** [reply, marked as answer]
hey @demolaf it's failing because of this error
`The provided entity includes an attribute with a value that has already been used`
The issue is that you are hardcoding the build name using `--build-name=0.0.1 `and that value, like the error message suggest, has already been utilized. 
Try to use it like I suggested above. This way, you can avoid conflicts with previously used build names.
```
flutter build ipa --release  \
      --build-name=1.0.$BUILD_NUMBER \
      --build-number=$BUILD_NUMBER \
```

**Reply by demolaf** [reply]
Ah okay makes sense now but this would create multiple versions for each build instead of filing the build (incremented) under the same version.
Correct me if I'm wrong though.

Thanks for the assistance, appreciate.

<img width="551" alt="Screenshot 2023-10-03 at 14 12 51" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/48495111/cb1daf41-e731-4308-a673-30283c1257b9">

**Reply by dtrdic** [reply]
Yep, this is how it's supposed to work. Each successful build will have its own version, and the bundle version must be higher than the previously uploaded version. You can't overwrite the already uploaded bundle.

**Reply by demolaf** [reply]
Hmmm I see, so I've been doing it wrong, thank you! 
Let me try it again.

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
